### PR TITLE
[V1][Core] min_p sampling support

### LINKED ISSUE #13058: [V1][Help Wanted] Porting missing sampling parameters to V1
### Anything you want to discuss about vllm.

To switch the engine from V0 to V1, we need to comprehensively support the sampling parameters in https://github.com/vllm-project/vllm/blob/main/vllm/sampling_params.py

While most of the key parameters are already supported, some of them are missing:

TODO (help wanted):
- [x] `n` (parallel sampling) #10980  @afeldman-nm 
- [x] `guided_decoding` (structured decoding) #12388  @aarnphm 
- [x] `logit_bias` #13079 @houseroad 
- [x] `min_p` #13191 @AoyuQC
- [ ] `bad_words` (originally implemented via logits processor) #13376 @22quinn 
- [x] `allowed_token_ids` (originally implemented via logits processor) #13210 @houseroad 

Parameters that will not be supported in V1:
* best_of
* logits_processors

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by AoyuQC**
Introduce min_p support in v1 Sampler.

FIX #13058 (*link existing issues this PR will resolve*)

### CODE REVIEW

**Review by WoosukKwon — APPROVED** [review]
LGTM. Thanks for submitting the PR! Could you please rebase with the main?

@njhill would you be ok with merging the PR?

**Review by njhill — APPROVED** [review]
LGTM, thanks!

**Comment by AoyuQC** [comment]
> LGTM, thanks!

Hi thanks for assisting, I just solve the conflicts

File: vllm/v1/worker/gpu_input_batch.py [hunk 1]
```diff
@@ -348,12 +365,13 @@ def condense(self, empty_req_indices: List[int]) -> None:
...
-                empty_index] = self.presence_penalties_cpu[last_req_index]
-            self.repetition_penalties_cpu[
-                empty_index] = self.repetition_penalties_cpu[last_req_index]
+            self.min_p_cpu[empty_index] = self.min_p_cpu[last_req_index]
+            self.frequency_penalties_cpu[empty_index] = \
+                self.frequency_penalties_cpu[last_req_index]
+            self.presence_penalties_cpu[empty_index] = \
+                self.presence_penalties_cpu[last_req_index]
+            self.repetition_penalties_cpu[empty_index] = \
+                self.repetition_penalties_cpu[last_req_index]
```
Comment (WoosukKwon) [inline vllm/v1/worker/gpu_input_batch.py:374]: nit: Could you please remove the unrelated changes?

Comment (AoyuQC) [reply vllm/v1/worker/gpu_input_batch.py:374]: > nit: Could you please remove the unrelated changes?

It comes from solving the conflicts

**Review by WoosukKwon — APPROVED** [review]
LGTM. Thanks!

File: vllm/v1/sample/sampler.py [hunk 2]
```diff
@@ -93,6 +93,10 @@ def sample(
             sampling_metadata.no_top_p,
             sampling_metadata.top_p,
         )
+
+        if not sampling_metadata.no_min_p:
+            logits = self.apply_min_p(logits, sampling_metadata.min_p)
```
Comment (njhill) [inline vllm/v1/sample/sampler.py:98]: @AoyuQC @WoosukKwon sorry I didn't look at this close enough before and just noticed this is wrong. It needs to be done before sampling of course or has no effect :)

I have included a fix in https://github.com/vllm-project/vllm/pull/13311

Comment (WoosukKwon) [reply vllm/v1/sample/sampler.py:98]: Oh you're right 😓 I didn't take a close look. Thanks for catching it!

**Comment by WoosukKwon** [comment]
Oops actually this PR broke v1-test. I ran the new test locally and thought it's ok. Let me fix it rn.
